src/Functions/DateTimeTransforms.h

```diff
@@ -1425,19 +1428,49 @@ struct ToDateTimeComponentsImpl
     using FactorTransform = ZeroTransform;
 };
 
+struct DateTimeAccurateConvertStrategyAdditions {};
+struct DateTimeAccurateOrNullConvertStrategyAdditions {};
 
-template <typename FromType, typename ToType, typename Transform, bool is_extended_result = false>
+template <typename FromType, typename ToType, typename Transform, bool is_extended_result = false, typename Additions = void *>
 struct Transformer
 {
     template <typename FromTypeVector, typename ToTypeVector>
-    static void vector(const FromTypeVector & vec_from, ToTypeVector & vec_to, const DateLUTImpl & time_zone, const Transform & transform)
+    static void vector(const FromTypeVector & vec_from, ToTypeVector & vec_to, const DateLUTImpl & time_zone, const Transform & transform,
+        ColumnUInt8::Container * vec_null_map_to [[maybe_unused]])
     {
         using ValueType = typename ToTypeVector::value_type;
         size_t size = vec_from.size();
         vec_to.resize(size);
 
         for (size_t i = 0; i < size; ++i)
         {
+            if constexpr (std::is_same_v<Additions, DateTimeAccurateConvertStrategyAdditions>
```

Comment:
I think the code would be ~less confusing~ easier to read if it was organized like this:

```cpp
if constexpr (To is a Date || To is a DateTime)
{
    if constexpr (Additions is the accurate strategy || Additions is the null strategy)
    {
        const bool is_valid_input = vec_from[i] ...
        if (!is_valid_input)
            ...
    }
}
```

Reply:
Fixed (reorganized)